### PR TITLE
Add GD5F QSPI driver

### PR DESCRIPTION
**Pull request by Jun-Tao**

### CODE REVIEW

File: drivers/mtd/spi_flash.c [hunk 1]
```diff
@@ -937,6 +937,14 @@ FAR struct mtd_dev_s *spi_initialize(int bus)
     }
 #endif
 
+#ifdef CONFIG_MTD_GD5F_QSPI
+    ret = gd5f_qspi_flash_initialize(priv);
+    if (ret == OK) {
+       ferr("GD5F QSPI flash is found\n");
```
Comment (haiyangyuneec) [inline drivers/mtd/spi_flash.c]: Please use finfo as above code.

File: drivers/mtd/gd5f_qspi.c [hunk 2]
```diff
@@ -0,0 +1,800 @@
...
+
+    ret = gd5f_readid(priv);
+    if (ret != OK) {
+        /* Unrecognized! Discard all of that work we just did and return NULL */
+
+        ferr("Unrecognized QSPI device\n");
+        goto errout;
+    }
+
+    /*Enable QE, then the quad io operations can be executed.*/
```
Comment (haiyangyuneec) [inline drivers/mtd/gd5f_qspi.c]: What does "QE" mean?

**Review by haiyangyuneec — CHANGES_REQUESTED** [review]

File: drivers/mtd/spi_flash.c (hunk 1 above)
Comment (Jun-Tao) [reply drivers/mtd/spi_flash.c]: OK.

File: drivers/mtd/gd5f_qspi.c (hunk 2 above)
Comment (Jun-Tao) [reply drivers/mtd/gd5f_qspi.c]: Quad IO Enable bit in register 0xB0.

**Review by haiyangyuneec — APPROVED** [review]

**Pull request merged**